File: elixir/word-count/word_count.exs
defmodule Words do
  def count(sentence) do
    sentence
    |> clean_words_and_list
    |> Enum.reduce(%{}, fn(key, counter) -> Map.update(counter, key, 1, &(&1 + 1)) end)
  end

  defp clean_words_and_list(sentence) do
    sentence
    |> String.downcase
    |> String.split(~r/[^[:alnum:]-]/u, trim: true)
  end
end

File: javascript/word-count/word-count.js
var Words = function() {};

Words.prototype.count = function(sentence) {
  var words = sentence.trim().replace(/ +/, " ").replace(/\t|\n/, " ").toLowerCase().split(" ");
  var counter = {};
  words.forEach(function(word) {
    if ((!counter[word]) || (word == 'constructor')) {
      counter[word] = 1;
    } else {
      counter[word]++;
    }
  });
  return counter;
};

module.exports = Words;
